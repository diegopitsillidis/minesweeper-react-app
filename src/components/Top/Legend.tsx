
import React, { FC } from 'react';
import styled from '@emotion/styled';

export interface LegendProps {
    /**
     * Feature that should be activated after first+second actions
     */
    feature: string;
    /**
     * First action
     */
    firstAction: string;
}

export const Legend: FC<LegendProps> = ({
    feature,
    firstAction
}) => (
    <Parent>
        <strong>{feature}: </strong>
        <FlagComboParent>
            <FirstAction>{firstAction}</FirstAction>
        </FlagComboParent>
    </Parent>
)

const FlagComboParent = styled.code`
    background: #e3e3e3;
`;

const Parent = styled.legend`
    font-size: 1em;
    margin: 0 auto 2vw;
    line-height: 1.25;
`;

const FirstAction = styled.span`
    color: #ec433c;
`;