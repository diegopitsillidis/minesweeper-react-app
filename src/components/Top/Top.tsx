import React, { FC } from 'react';
import styled from '@emotion/styled';

import { Legend, LegendProps} from "./Legend";
import { GameName, GameNameProps } from "./GameName";

// interface TopProps {}
export type TopComponentType = LegendProps & GameNameProps

export const Top: FC<TopComponentType> = ({
    children, 
    ...LegendProps
}) => (
    <Header>
        <GameName>{children}</GameName>
        <Legend {...LegendProps} />
    </Header>
);

const Header = styled.header`
    text-align: center;
    position: relative;
    display: inline-block;
`;