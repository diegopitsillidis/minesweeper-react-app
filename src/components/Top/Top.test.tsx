import React from 'react';
import { render } from "@testing-library/react";
import { Top } from './Top';

it('Legend renders correctly', () => {
    const args = {
        children: 'Minesweeper',
        feature: 'Flag',
        firstAction: 'Right Click',
    }

    const { asFragment } = render(<Top {...args} />);

    expect(asFragment()).toMatchSnapshot();
});