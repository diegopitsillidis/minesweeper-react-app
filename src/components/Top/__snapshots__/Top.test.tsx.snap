// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Legend renders correctly 1`] = `
<DocumentFragment>
  <header
    class="css-1mm4p1j"
  >
    <h1
      class="css-aqrpco"
    >
      Minesweeper
    </h1>
    <legend
      class="css-y5wpvr"
    >
      <strong>
        Flag: 
      </strong>
      <code
        class="css-1th0bce"
      >
        <span
          class="css-1eygtyr"
        >
          Right Click
        </span>
      </code>
    </legend>
  </header>
</DocumentFragment>
`;
